### PR TITLE
Feature/embeded structs example

### PR DESCRIPTION
**Pull request by mvo5**
This branch adds a small example how to decode embedded structs to help users of the library to find this nice feature more easily.

### CODE REVIEW

**Comment by kentoj** [comment]
+1
I have an example that parses a map from an embedded structure. I can add that later as well.

File: yaml.go [hunk 1]
```diff
@@ -237,6 +237,11 @@ func getStructInfo(st reflect.Type) (*structInfo, error) {
 		}
 
 		inline := false
+		// a embedded struct is auto-inlined, see LP: #1191814
+		if field.Anonymous {
+			inline = true
```
Comment (niemeyer) [inline yaml.go]: This doesn't seem related to the patch description.

File: decode_test.go [hunk 2]
```diff
@@ -936,6 +936,21 @@ func (s *S) TestUnmarshalSliceOnPreset(c *C) {
 	c.Assert(v.A, DeepEquals, []int{2})
 }
 
+func (s *S) TestUnmarshalEmbeddedStructs(c *C) {
```
Comment (niemeyer) [inline decode_test.go]: This also seems related to the unwanted change rather than the example.

**Comment by mvo5** [comment]
Hey Gustavo, I'm sorry for this mistake. I re-pushed with only the relevant revision.

**Review by rogpeppe — APPROVED** [review]
LGTM, thanks.
